TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/cli.py

```diff
@@ -93,42 +98,75 @@ def start_replayer_cmd(ctx):
 def backfill_group(ctx):
     """All actions related to historical/backfill data migrations"""
     if ctx.env.backfill is None:
-        raise ValueError("Backfill migration is not set")
+        raise click.UsageError("Backfill migration is not set")
+
 
+@backfill_group.command(name="describe")
+@click.pass_obj
+def describe_backfill_cmd(ctx):
+    click.echo(logic_backfill.describe(ctx.env.backfill, as_json=ctx.json))
 
-@backfill_group.command(name="create-migration")
+
+@backfill_group.command(name="create")
 @click.option('--pipeline-template-file', default='/root/osiPipelineTemplate.yaml', help='Path to config file')
 @click.option("--print-config-only", is_flag=True, show_default=True, default=False,
               help="Flag to only print populated pipeline config when executed")
 @click.pass_obj
-def create_migration_backfill_cmd(ctx, pipeline_template_file, print_config_only):
-    """Create migration action"""
-    ctx.env.backfill.create(pipeline_template_path=pipeline_template_file, print_config_only=print_config_only)
+def create_backfill_cmd(ctx, pipeline_template_file, print_config_only):
+    exitcode, message = logic_backfill.create(ctx.env.backfill,
+                                              pipeline_template_path=pipeline_template_file,
+                                              print_config_only=print_config_only)
+    if exitcode != 0:
+        raise click.ClickException(message)
+    click.echo(message)
 
 
-@backfill_group.command(name="start-migration")
+@backfill_group.command(name="start")
 @click.option('--pipeline-name', default=None, help='Optionally specify a pipeline name')
 @click.pass_obj
-def start_migration_backfill_cmd(ctx, pipeline_name):
-    """Start migration action"""
-    ctx.env.backfill.start(pipeline_name=pipeline_name)
+def start_backfill_cmd(ctx, pipeline_name):
+    exitcode, message = logic_backfill.start(ctx.env.backfill, pipeline_name=pipeline_name)
+    if exitcode != 0:
+        raise click.ClickException(message)
+    click.echo(message)
 
 
-@backfill_group.command(name="stop-migration")
+@backfill_group.command(name="stop")
 @click.option('--pipeline-name', default=None, help='Optionally specify a pipeline name')
 @click.pass_obj
-def stop_migration_backfill_cmd(ctx, pipeline_name):
-    """Stop migration action"""
-    ctx.env.backfill.stop(pipeline_name=pipeline_name)
+def stop_backfill_cmd(ctx, pipeline_name):
+    exitcode, message = logic_backfill.stop(ctx.env.backfill, pipeline_name=pipeline_name)
+    if exitcode != 0:
+        raise click.ClickException(message)
```

Comment:
Not a blocker, just curious - why raise ClickException rather than anything else?

Reply:
It exits with a non-zero exit code and also displays the error "nicely" --not with a full traceback, etc. I think subclassing that exception to create custom ones could be a good way to go in the future to get more specific errors while keeping the convenience features.

Docs are here: https://click.palletsprojects.com/en/8.1.x/exceptions/#exception-handling